.cirrus.yml

```diff
@@ -145,12 +145,24 @@ task:
 
     - name: customer_testing-linux
       only_if: "$CIRRUS_PR != ''"
-      # environment:
+      environment:
         # Empirically, this shard runs fine at 1 CPU and 4G RAM as of October 2019. We will probably
         # want to grow this container when we invite people to add their tests in large numbers.
+        SHOULD_UPDATE_PACKAGES: "FALSE"
       script:
+        # This should exactly match the contents of
+        # https://flutter.googlesource.com/recipes/+/refs/heads/master/recipe_modules/adhoc_validation/resources/customer_testing.sh
+        #
+        # We don't run "flutter update-packages", but we do need these two directories updated:
+        - (cd dev/tools; pub get)
+        - (cd dev/customer_testing; pub get)
+        # Cirrus doesn't give us the master branch, so we have to fetch it for ourselves,
+        # otherwise we won't be able to figure out how old or new our current branch is.
+        - git fetch origin master:master
```

Comment:
Can the shared logic with luci be moved into a separate script? If that's available, i'll happily send a recipes CL so we no longer need to rely on the adhoc script there.

Reply:
Possibly. The logic here is sort of Cirrus-specific because Cirrus does weird stuff with the checkout, no idea what the impact of running those steps on LUCI would be.

Reply:
That makes sense. In general, infra logic always handles the source checkout. Whereas

```
        - (cd dev/tools; pub get)
        - (cd dev/customer_testing; pub get)
        - rm -rf bin/cache/pkg/tests
        - git clone https://github.com/flutter/tests.git bin/cache/pkg/tests
        - git -C bin/cache/pkg/tests checkout `dart --enable-asserts dev/tools/bin/find_commit.dart bin/cache/pkg/tests`
        - dart --enable-asserts dev/customer_testing/run_tests.dart --skip-on-fetch-failure --skip-template bin/cache/pkg/tests/registry/*.test
 ```
look like they can be shared